sources/Outputs/include/Job.h

```diff
@@ -37,13 +39,60 @@ class Job {
     /**
      * @brief Constructor
      *
-     * @param filepath output filename
+     * @param filename output filename
+     * @param lvl dynawo log level
+     */
+    JobDefinition(const std::string& filename, const std::string& lvl) : filename(filename), dynawoLogLevel(lvl), contingencyId{}, baseFilename{} {}
+
+    /**
+     * @brief Constructor
+     *
+     * @param filename output filename
      * @param lvl dynawo log level
+     * @param config configuration
      */
-    JobDefinition(const std::string& filepath, const std::string& lvl) : filename(filepath), dynawoLogLevel(lvl) {}
+    JobDefinition(const std::string& filename, const std::string& lvl, const dfl::inputs::Configuration& config) :
+        filename(filename),
+        dynawoLogLevel(lvl),
+        contingencyId{},
+        baseFilename{} {
+      initFromConfig(config);
+    }
 
-    std::string filename;        ///< filename of the job output file
-    std::string dynawoLogLevel;  ///< Dynawo log level, in string representation
+    /**
+     * @brief Constructor with a reference to a contingency
+     *
+     * @param filename output filename
+     * @param lvl dynawo log level
+     * @param config configuration
+     * @param contingencyId contingency identifier
+     * @param baseFilename filename of base case
+     */
+    JobDefinition(const std::string& filename, const std::string& lvl, const dfl::inputs::Configuration& config, const std::string& contingencyId,
+                  const std::string& baseFilename) :
+        filename(filename),
+        dynawoLogLevel(lvl),
+        contingencyId(contingencyId),
+        baseFilename(baseFilename) {
+      initFromConfig(config);
+    }
+
+    std::string filename;                ///< filename of the job output file
+    std::string dynawoLogLevel;          ///< Dynawo log level, in string representation
+    std::string contingencyId;           ///< Identifier of referred contingency
+    std::string baseFilename;            ///< Name for base case filename if we are defining a jobs file for a contingency
+    std::chrono::seconds startTime{0};   ///< The start time of the simulation
+    std::chrono::seconds stopTime{100};  ///< the constant duration of the simulation in the job
+
+    /**
+     * @brief Initialize additional attributes from configuration
+     *
+     * @param config configuration
+     */
+    void initFromConfig(const dfl::inputs::Configuration& config) {
```

Comment:
Good practices: The declaration order in classes/structures is TYPES, FUNCTIONS, MEMBERS

Reply:
Done, `initFromConfig` method moved before members of structure `JobDefinition`.

Reply:
Sorry I think what was missing at first is the key word private for initFromConfig function. The function should be declared private after all public functions and members

Reply:
You are right. Declared the method in the private section, after public functions and members.